diff -urb js/ui/background.js js2/ui/background.js
--- a/js/ui/background.js	2019-02-10 22:03:17.160835583 +0200
+++ b/js/ui/background.js	2019-02-10 22:05:36.123342025 +0200
@@ -145,6 +145,7 @@
         this._fileMonitors = {};
         this._backgroundSources = {};
         this._animations = {};
+        this._clock = null;
     },
 
     monitorFile(file) {
@@ -165,6 +166,12 @@
         this._fileMonitors[key] = monitor;
     },
 
+    getWallClock: function() {
+        if (!this._clock)
+            this._clock = new GnomeDesktop.WallClock();
+        return this._clock;
+    },
+
     getAnimation(params) {
         params = Params.parse(params, { file: null,
                                         settingsSchema: null,
@@ -252,7 +259,8 @@
         this._cancellable = new Gio.Cancellable();
         this.isLoaded = false;
 
-        this._clock = new GnomeDesktop.WallClock();
+        let cache = getBackgroundCache()
+        this._clock = cache.getWallClock();
         this._timezoneChangedId = this._clock.connect('notify::timezone',
             () => {
                 if (this._animation)
