diff -urb js/ui/sessionMode.js js.b/ui/sessionMode.js
--- js/ui/sessionMode.js	2017-10-20 14:14:09.664870517 +0300
+++ js.b/ui/sessionMode.js	2017-10-19 10:08:21.407848262 +0300
@@ -45,6 +45,7 @@
         hasNotifications: true,
         isGreeter: true,
         isPrimary: true,
+        allowExtensions: true,
         unlockDialog: imports.gdm.loginDialog.LoginDialog,
         components: ['polkitAgent'],
         panel: {
@@ -50,7 +50,7 @@
         panel: {
             left: [],
             center: ['dateMenu'],
-            right: ['a11yGreeter', 'keyboard', 'aggregateMenu'],
+            right: ['keyboard', 'aggregateMenu'],
         },
         panelStyle: 'login-screen'
     },
@@ -59,6 +60,7 @@
         isLocked: true,
         isGreeter: undefined,
         unlockDialog: undefined,
+        allowExtensions: true,
         components: ['polkitAgent', 'telepathyClient'],
         panel: {
             left: [],
@@ -71,6 +73,7 @@
     'unlock-dialog': {
         isLocked: true,
         unlockDialog: undefined,
+        allowExtensions: true,
         components: ['polkitAgent', 'telepathyClient'],
         panel: {
             left: [],
